### ISSUE TITLE
max_priority parameter is ignored in getBakingRights when it's value is zero

### ISSUE THREAD

**Issue by itkach** [ecadlabs/taquito#580]
Result of calling RPC `getBakingRights` with `max_priority` parameter set to `0` like so 
```
await rpc.getBakingRights(
      {
        max_priority: 0,
        cycle,
        delegate: baker
      },
      { block }
    )
```
is unfiltered list of baking rights with all possible priority values. Instead, i should only include rights with priority 0

HTTP URL Taquito generates looks like this:

```
https://mainnet-tezos.giganode.io/chains/main/blocks/BL6Bzq85kgRBsr9N5koyZCGZtbVJ9TvgApDhHRTBjX64eeya77d/helpers/baking_rights?cycle=317&delegate=tz3UoffC7FG7zfpmvmjUmUeAaHvzdcUvAj6r
```

Note that `max_priority` is omitted

To compare, when max_priority is 1 URL looks like this:

```https://mainnet-tezos.giganode.io/chains/main/blocks/BLo36ctsfF35XXM4raYcTEf1m7bxC1e4RBX21343MaPJPmU24Ax/helpers/baking_rights?max_priority=1&cycle=317&delegate=tz3UoffC7FG7zfpmvmjUmUeAaHvzdcUvAj6r```

This is with taquito 7.1.0-beta.0

**Comment by itkach** [comment]
Possibly the issue is here: https://github.com/ecadlabs/taquito/blob/66c1c9fb402d4589c7ccb06a34465420b996d416/packages/taquito-http-utils/src/taquito-http-utils.ts#L45-L49
`0` is false-y so is incorrectly serialized as empty string

**Comment by jevonearth** [comment]
@itkach This issue was addressed in `8.0.2-beta.0`. Do let us know if you have any problems. Thanks for the report!
